src/workload.py

```diff
@@ -87,27 +87,15 @@ def healthy(self) -> bool:
         if not self.alive:
             return False
 
-        # netcat isn't a default utility, so can't guarantee it's on the charm containers
-        # this ugly hack avoids needing netcat
-        bash_netcat = (
-            f"echo '4lw' | (exec 3<>/dev/tcp/localhost/{CLIENT_PORT}; cat >&3; cat <&3; exec 3<&-)"
-        )
-        ruok = [bash_netcat.replace("4lw", "ruok")]
-        srvr = [bash_netcat.replace("4lw", "srvr")]
+        try:
+            response = json.loads(
+                self.exec(["curl", f"localhost:{ADMIN_SERVER_PORT}/commands/ruok", "-m", "10"])
+            )
 
-        # timeout needed as it can sometimes hang forever if there's a problem
-        # for example when the endpoint is unreachable
-        timeout = ["timeout", "10s", "bash", "-c"]
+        except (ExecError, CalledProcessError, json.JSONDecodeError):
+            return False
 
-        try:
-            ruok_response = self.exec(command=timeout + ruok)
-            if not ruok_response or "imok" not in ruok_response:
-                return False
-
-            srvr_response = self.exec(command=timeout + srvr)
-            if not srvr_response or "not currently serving requests" in srvr_response:
-                return False
-        except (ExecError, CalledProcessError):
+        if response.get("error", None) is not None:
```

Comment:
Simplified the condition in 354e40a12ff8. The output changes a bit from the 4 letters commands though, here a "imok" response looks like this (with a HTTP 200 status):
```json
{
  "command" : "ruok",
  "error" : null
}
```
We do not need the `srvr` command either, because the `"not currently serving requests"` message would appear in the "error field"